src/adapters/cloudflare/mod.rs

```diff
@@ -65,8 +66,104 @@ impl CloudFlareClient {
+        worker_version_id: String,
+        status_code_range_start: u16,
+        status_code_range_end: u16,
+        from_time: chrono::DateTime<chrono::Utc>,
+        to_time: chrono::DateTime<chrono::Utc>,
+    ) -> Result<u32> {
+        let path = format!("/accounts/{account_id}/workers/observability/telemetry/query");
+        let url = Self::url_with_path(&path);
+
+        // Convert DateTime to Unix timestamps
+        let from_timestamp = from_time.timestamp() as u64;
+        let to_timestamp = to_time.timestamp() as u64;
+
+        let query_body = serde_json::json!({
+            "view": "calculations",
+            "queryId": "worker-calculation",
+            "parameters": {
+                "datasets": ["cloudflare-workers"],
+                "filters": [
+                    {
+                        "key": "$metadata.service",
+                        "operation": "eq",
+                        "type": "string",
+                        "value": worker_name
+                    },
+                    {
+                        "key": "$workers.scriptVersion.id",
+                        "operation": "eq",
+                        "type": "string",
+                        "value": worker_version_id
+                    },
+                    {
+                        "key": "$workers.event.response.status",
+                        "operation": "gte",
+                        "type": "number",
+                        "value": status_code_range_start
+                    },
+                    {
+                        "key": "$workers.event.response.status",
+                        "operation": "lte",
+                        "type": "number",
+                        "value": status_code_range_end
+                    }
+                ],
+                "calculations": [
+                    {
+                        "key": "$workers.event.response.status",
+                        "operator": "count",
+                        "keyType": "number",
+                        "alias": "sum"
+                    }
+                ]
+            },
+            "timeframe": {
+                "to": to_timestamp,
+                "from": from_timestamp
+            }
+        });
+
+        let response = self
+            .client
+            .post(url)
+            .json(&query_body)
+            .send()
+            .await
+            .into_diagnostic()?;
+
+        // If there's an error, just return 0 results
+        if !response.status().is_success() {
+            debug!(
+                "Failed to query Cloudflare metrics for worker: {}, version: {}, status codes: {}, error: {}",
+                worker_name,
+                worker_version_id,
+                status_code_range_start,
+                response
+                    .json()
+                    .await
+                    .unwrap_or_else(|_| "Unknown error".to_string())
+            );
+            return Ok(0);
+        }
```

Comment:
I think this has to be a higher log level so users can know when something is failing. Otherwise they're just going to be sitting waiting for the release, but no error message will show on screen.
```suggestion
        if !response.status().is_success() {
            error!(
                "Failed to query Cloudflare metrics for worker: {}, version: {}, status codes: {}, error: {}",
                worker_name,
                worker_version_id,
                status_code_range_start,
                response
                    .json()
                    .await
                    .unwrap_or_else(|_| "Unknown error".to_string())
            );
            return Ok(0);
        }
```

Reply:
Fixed and updated the requisite Cloudwatch one, too